nginx/Dockerfile

```diff
@@ -0,0 +1,18 @@
+# Inspired by https://medium.com/pernod-ricard-tech/adding-basic-authentication-with-nginx-as-a-reverse-proxy-a229f9d12b73
+
+FROM nginx:latest
+
+RUN apt-get update -y && apt-get install -y apache2-utils && rm -rf /var/lib/apt/lists/*
+
+# Basic auth credentials
+ENV BASIC_AUTH_USERNAME=airbyte
+ENV BASIC_AUTH_PASSWORD=password
```

Comment:
Currently the username and password env variables are set in both here and the `docker-compose` file. Will the one in `docker-compose` override them here? If not, to change the password, users will have to update them in both places. Ideally they should only need to do it in one place.

Maybe something like the following:

```
ARG BASIC_AUTH_USERNAME
ARG BASIC_AUTH_PASSWORD

ENV BASIC_AUTH=${BASIC_AUTH}
ENV BASIC_AUTH_PASSWORD=${BASIC_AUTH_PASSWORD}
```

I am not 100% sure this is correct, but the idea is to:
- Set the variables only in `docker-compose`.
- Pass them from `docker-compose` to the image during build time as `build-arg`.

Reply:
If the above approach works, it's better to use different names to avoid confusion:

```dockerfile
ARG AIRBYTE_LOGIN_USERNAME
ARG AIRBYTE_LOGIN_PASSWORD

ENV BASIC_AUTH=${AIRBYTE_LOGIN_USERNAME}
ENV BASIC_AUTH_PASSWORD=${AIRBYTE_LOGIN_PASSWORD}
```

The `AIRBYTE_LOGIN_USERNAME` and `AIRBYTE_LOGIN_PASSWORD` are set in `docker-compose`. They are more user friendly, and are the ones that users should change.

Reply:
I *think* the convention is to use the same name so they can be find/replaced simply.  The reason for defining the ENV in both the Dockerfile and the docker-compose is that that the Dockerfile sets defaults, so the container can be run outside of the docker-compose orchestrator (e.g. in K8s)

Reply:
Update - you were right! I moved things into the configs/env files, and the Dockerfile itself is now setting empty values.  Now folks can toggle the feature on and off.